In [1]:
import numpy as np

Определяем функции для подсчета размещений, перестановок и сочетаний

In [4]:
def combinations(n: int, k: int) -> int:
    """Число сочетаний.
    """
    
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

### Задача 1

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

### Решение: 
пусть A - событие при котором в вытянутых картах есть хотя бы 1 туз, а B - событие при котором в вытянутых картах нет ни одного туза. Тогда P(A)=1-P(B)
P(B) = (количество вариантов, когда среди вытянутых 4 карт нет ни одного туза)/(количество вариантов вытянуть 4 карты из полной колоды)

Количество способов вытащить 4 карты из колоды без тузов = 𝑛!/𝑘!(𝑛−𝑘)!, где n=48 (n- кол-во карт без тузов), а k=4 (k- кол-во карт)

In [5]:
combinations(48,4)

194580

Количество вариантов вытянуть 4 карты из полной колоды:

In [6]:
combinations(52,4)

270725

Количество исходов при которых может  быть вытянут хотя бы 1 туз:

In [7]:
combinations(52,4)-combinations(48,4)

76145

Ответ: 76145

### Задача 2

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что рядом будут сидеть: а) два определённых человека? б) три определённых человека?

#### Решение
Возможное количество вариантов рассадки 7 человек = 7!
Когда на скамейку сажаем 7 человек, то 2 человека рядом могут разместиться 2-мя способами (один слева , другой справа и наоборот), а оставшиеся 5 человек могут разместиться 5! способами. Итого 2 * 5! для одного варианта размещения, когда, например, пара находится с одного края скамейки. Всего таких вариантов размещения пары 6: 

1: 12XXXXX
2: X12XXXX
3: XX12XXX
4: XXX12XX
5: XXXX12X
6: XXXXX12

Итого количество исходов для размещения 2 человек = 6 * 2 * 5! 

Вероятность того, что рядом будут сидеть 2 человека: 6 * 2 * 5! / 7! = 2/7

Если 3 человека то они могут разместиться рядом 3!=6 способами. Оставшиеся 4 разместятся тогда 4! способами. 
Итого 3! * 4! способов размещения для одного варианта расположения. Таких вариантов расположения 5:
    1: 123XXXX
    2: X123XXX
    3: XX123XX
    4: XXX123X
    5: XXXX123
        
Итого количество исходов для размещения 3 человек = 6 * 4! * 5 

Вероятность того, что рядом будут сидеть 3 человека: 6 * 4! * 5 / 7! = 1/7

Ответ: для 2-х человек - 2/7, для 3 человек - 1/7

### Задача 3

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. 
Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

#### Решение

Количество способов которыми можно выбрать 3 вопроса из 60: 

In [8]:
combinations(60,3)

34220

Студент знает 50 вопросов. Значит количество способов, когда студент из трех выбранных вопросов знает все составляет:

In [9]:
combinations(50,3)

19600

А  количество способов, когда студент знает только 2 вопроса из трех- это произведение количества способов когда студент знает 2 вопроса на количество способов, при которых студент вытянет 1 неудачный вопрос (из 10)

In [12]:
combinations(50,2)*combinations(10,1)

12250

Итак ответ: вероятность когда знает ответы на все вопросы P(A)=19600 / 34220 = 0.573 , а вероятность того, что знает только 2 вопроса: 
    P(B) = 12250 / 34220 = 0.358

### Задача 4

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

#### Решение

Для независимых событий должно выполняться равенство P(AB)=P(A)*P(B)
Варианты события А : 2,4,6
Варианты события B: 4,5,6
Вероятность события когда выпало четное число больше трех ("4" или "6") -это 2 варианта из 6 возможных: 2 / 6 = 1 / 3
Вероятность события А составляет 0.5 ， вероятность события В тоже составляет 0.5
P(A)*P(B)=0.25
Т.о. равенство P(AB)=P(A)*P(B) не соблюдается. События не являются независимыми.

### Задача 5

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

### Решение

P(A)- вероятность, что болен =0.01 (т.к. поражает 0.1% населения), 
P(B) -вероятность, что тест верный = 0.99 . 
P(-A) - вероятность, что не болен = 1 - P(A) = 0.99
P(C) - вероятность, что тест ложный = 0.01 .
По формуле полной вероятности, тест будет положительным с вероятностью: P(o) = 0.01 * 0.99 + 0.99 * 0.1
По формуле Байеса вероятность, что действительно болен 

P(A) * P(B) / P(o) = 0.5
